ShowerShow/Control/ShowerThoughtController.cs

```diff
@@ -0,0 +1,196 @@
+
+                //this is to give priority to tasks
+                Task getId = Task.Run(() =>
+                {
+                    thought = showerThoughtService.GetShowerThoughtById(ThoughtId).Result;
+                });
+                await getId.ContinueWith(prev =>
+                {
+                    showerThoughtService.DeleteShowerThought(thought);
+                });
+                responseData.StatusCode = HttpStatusCode.OK;
+                await responseData.WriteAsJsonAsync(thought);
+                return responseData;
+
+            }
+            catch
+            {
+                responseData.StatusCode = HttpStatusCode.BadRequest;
+                return responseData;
+            }
+        }
+        [Function("GetThoughtById")]
+        [OpenApiOperation(operationId: "GetThoughtById", tags: new[] { "ShowerThoughts" })]
+        [OpenApiParameter(name: "ThoughtId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The thought ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(ShowerThought), Description = "The OK response with the retrieved thought")]
+        public async Task<HttpResponseData> GetThoughtById([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "shower/thoughts/{ThoughtId:Guid}")] HttpRequestData req, Guid ThoughtId)
+        {
+            _logger.LogInformation("Retrieving schedule.");
+
+
+            HttpResponseData responseData = req.CreateResponse();
+            try
+            {
+                ShowerThought thought = await showerThoughtService.GetShowerThoughtById(ThoughtId);
+                await responseData.WriteAsJsonAsync(thought);
+                responseData.StatusCode = HttpStatusCode.OK;
+                return responseData;
+
+            }
+            catch
+            {
+                responseData.StatusCode = HttpStatusCode.BadRequest;
+                return responseData;
+            }
+        }
+        [Function("GetThoughtByShowerId")]
+        [OpenApiOperation(operationId: "GetThoughtByShowerId", tags: new[] { "ShowerThoughts" })]
+        [OpenApiParameter(name: "ShowerId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The shower ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(ShowerThought), Description = "The OK response with the retrieved thoughts")]
+        public async Task<HttpResponseData> GetThoughtByShowerId([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "shower/thoughts/{ShowerId:Guid}/s")] HttpRequestData req, Guid ShowerId)
+        {
+            _logger.LogInformation("Retrieving schedule.");
+
+
+            HttpResponseData responseData = req.CreateResponse();
+            try
+            {
+                ShowerThought thought = await showerThoughtService.GetThoughtByShowerId(ShowerId);
```

Comment:
Can you add check if user exist and active here too. Ideally all gets should have it

Reply:
Also you'd need a check where shower id is correct as well right?